re.go

```diff
@@ -29,6 +29,14 @@ import (
 	"strconv"
 )
 
+// A pointer to a Position can be passed as the third argument to Scan to
+// record position information for the match.  This can be used in a loop to
+// continue scanning from the end of previous first match.
```

Comment:
I'm essentially proposing an API where there:

1. 2 required arguments: needle (the regexp) and haystack
2. An optional argument (a Range describing the whole match)
3. Variable arguments that come after the optional argument.

Of course, Go doesn't suppport optional arguments, but we can fake it since our variable arguments are interface{}

Reply:
OK. Done.